storage.go

```diff
@@ -508,18 +549,21 @@ func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int6
 	return size, nil
 }
 
-func (s *Storage) writeMultipart(ctx context.Context, o *Object, r io.Reader, size int64, index int, opt pairStorageWriteMultipart) (n int64, err error) {
+func (s *Storage) writeMultipart(ctx context.Context, o *Object, r io.Reader, size int64, index int, opt pairStorageWriteMultipart) (n int64, part *Part, err error) {
 	if o.Mode&ModePart == 0 {
-		return 0, services.ObjectModeInvalidError{Expected: ModePart, Actual: o.Mode}
+		return 0, nil, services.ObjectModeInvalidError{Expected: ModePart, Actual: o.Mode}
 	}
 
 	input := &s3.UploadPartInput{
-		Bucket:        &s.name,
-		PartNumber:    aws.Int64(int64(index)),
+		Bucket: &s.name,
+		// For S3, the `PartNumber` is [1, 10000]. But for users, the `PartNumber` is zero-based.
+		// Set PartNumber=index+1 here to ensure pass in an effective `PartNumber` for `UploadPart`.
+		// ref: https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html
+		PartNumber:    aws.Int64(int64(index + 1)),
 		Key:           aws.String(o.ID),
 		UploadId:      aws.String(o.MustGetMultipartID()),
 		ContentLength: &size,
-		Body:          iowrap.SizedReadSeekCloser(r, size),
+		Body:          aws.ReadSeekCloser(r),
```

Comment:
Why we need to migrate from `iowrap.SizedReadSeekCloser` to `aws.ReadSeekCloser`?

Reply:
I don't know why `WirteMultipart` got the ` RequestError` in the last commit. I found that `aws.ReadSeekCloser` is used in `write`, so I migrated to `aws.ReadSeekCloser` to have a try. The error info is:
```
RequestError: send request failed
...
net/http: HTTP/1.x transport connection broken: http: ContentLength=1543539 with Body length 0'
```

Reply:
Please figure why.

If there is a bug in iowrap, please fix it. Don't    solve problem by skip it.

Reply:
Maybe something buggy in https://github.com/aos-dev/go-storage/blob/master/pkg/iowrap/reader.go#L93-L108 ?

Reply:
S3 SDK will compute Content MD5 by default which will consume our input reader.

In go-storage, we will let users calculate content md5 and passed into as a pair `Content-MD5`.

So we need to set `S3DisableContentMD5Validation` to true:

```go
cfg.S3DisableContentMD5Validation = aws.Bool(true)
```

Reply:
Please change it back, we will use our own `iowrap` instead of the one provided by SDK.